installscripts/cookbooks/jenkins/recipes/configurejenkins.rb

```diff
@@ -69,27 +59,27 @@
 
 execute 'extractXmls' do
```

Comment:
Can you try moving this XMl extraction/copying section to the Dockerfile/`setupjenkins` script as well?

If you hit a snag with that, just add a TODO here mentioning that we ought to do that in the future.

Reply:
This is used in all scenarios. If we move this XML extraction to setupjenkins.rb (specifically for scenario1), then we need to handle the extract xml.tar for scenario2,3. Probably in chefclient.sh file.

Reply:
I was suggesting that we could just add a conditional around the existing XML stuff in `configurejenkins.rb` like you have already done elsewhere in the file, and only run that section for Scenario1. Then for scenario 2/3 we could handle it in the Dockerfile.

But maybe this ought to be done later, as part of a larger effort to stop using Chef entirely for everything except Scenario 1.